site/.eleventy.js

```diff
@@ -0,0 +1,181 @@
+        if (mainVersionCmp !== 0) {
+          return mainVersionCmp;
+        }
+        return parseInt(b.split(".")[1]) - parseInt(a.split(".")[1]);
+      })
+
+
+    allRows.slice(0, 5).forEach(version => { results[version] = releases[version] })
+
+    if (allRows.length > 5) {
+      results["Other"] = allRows
+        .slice(5)
+        .reduce((accumulator, item) => accumulator + releases[item], 0)
+    }
+
+    return results
+  });
+
+  eleventyConfig.addFilter("osVersionSummary", (data) => {
+    const results = {}
+
+    const sortedVersions = Object.keys(data).sort((a, b) => data[b] - data[a]);
+    sortedVersions.slice(0, 5).forEach(version => { results[version] = data[version] })
+
+    if (sortedVersions.length > 5) {
+      results["Other"] = sortedVersions
+        .slice(5)
+        .reduce((accumulator, currentValue) => accumulator + data[currentValue], 0)
+    }
+
+    return results
+  });
+
+  eleventyConfig.addFilter("sortBoards", (value) => {
+    const data = { data: [], labels: [], backgroundColor: [] }
+    Object.keys(value).sort().forEach(key => { data.labels.push(friendlyBoardName[key] || key); data.data.push(value[key]); data.backgroundColor.push(randomColor()) })
+    return data
+  });
+
+  eleventyConfig.addFilter("mergeInstallationHistory", (history) => {
+    const data = { total: [], os: [], core: [], supervised: [], container: [] }
+
+    for (const entry of historyFiltering(history)) {
+      data.total.push({ x: (entry.timestamp), y: entry.active_installations })
+      data.os.push({ x: (entry.timestamp), y: entry.installation_types.os })
+      data.core.push({ x: (entry.timestamp), y: entry.installation_types.core })
+      data.supervised.push({ x: (entry.timestamp), y: entry.installation_types.supervised })
+      data.container.push({ x: (entry.timestamp), y: entry.installation_types.container })
+    }
+
+    return JSON.stringify([
+      dataPoint({ data: data.total, label: "Total", borderColor: "#3366cc" }),
+      dataPoint({ data: data.os, label: "Operating System", borderColor: "#dc3912" }),
+      dataPoint({ data: data.container, label: "Container", borderColor: "#ff9900" }),
+      dataPoint({ data: data.core, label: "Core", borderColor: "#109618" }),
+      dataPoint({ data: data.supervised, label: "Supervised", borderColor: "#990099" })
+    ])
+  }
+  )
+
+  eleventyConfig.addFilter("mergeVersionHistory", (history) => {
+    const allVersions = new Set()
+    const versionHistoryData = historyFiltering(history)
+
+    for (const entry of versionHistoryData) {
+      Object.keys(entry.versions).forEach((version) =>
+        allVersions.add(version)
+      );
+    }
+
+    const versionsOrdered = Array.from(allVersions).sort((a, b) => {
+      const mainVersionCmp =
+        parseInt(b.split(".")[0]) - parseInt(a.split(".")[0]);
+      if (mainVersionCmp !== 0) {
+        return mainVersionCmp;
+      }
+      return parseInt(b.split(".")[1]) - parseInt(a.split(".")[1]);
```

Comment:
you could extract this version sorting.

Reply:
Yeah, this was just copy+paste from old site